File: Problem_80_Euler/main.py
from decimal import Decimal, localcontext

def prob80():
    total = 0
    for x in range(1,100):
        with localcontext() as ctx:
            ctx.prec = 105
            if len(str(Decimal(x).sqrt())) == 1:
                total+=0
            else:
                a = sum([int(i) for i in str(Decimal(x).sqrt())[2:101]])+int(str(Decimal(x).sqrt())[0])
                total+=a
    return total
print(prob80())

File: Problem_80_Euler/main.go
package main

import (
	"flag"
	"fmt"
	"math/big"
	"strings"
)

// https://projecteuler.net/problem=80

// Use main.py as golang has precision issues
func main() {
	window := flag.Int("window", 100, "The window size")
	flag.Parse()

	prec := uint(200)
	total := 0
	for i := 0; i <= *window; i++ {
		val := new(big.Float).Sqrt(big.NewFloat(float64(i)).SetPrec(100)).SetPrec(prec)
		newVal := fmt.Sprintf("%.100f", val)
		dValue := strings.Split(newVal, ".")[1]
		fmt.Println(dValue)
		for _, char := range strings.TrimSpace(dValue) {
			fmt.Println(string(char), int(char-'0'))
			total += int(char - '0')
		}
	}

	fmt.Println(total)
}
